ouroboros-consensus-cardano/test/shelley-test/Test/Consensus/Shelley/Serialisation.hs

```diff
@@ -29,14 +30,17 @@ import           Test.Util.Serialisation.Roundtrip
 
 tests :: TestTree
 tests = testGroup "Shelley"
-    [ roundtrip_all testCodecCfg dictNestedHdr
-
+    [ testGroup "Shelley examples roundtrip" $ examplesRoundtrip ShelleyCodecConfig Shelley.Examples.examplesShelley
+    , testGroup "Mary examples roundtrip"    $ examplesRoundtrip ShelleyCodecConfig Shelley.Examples.examplesMary
+    , testGroup "Allegra examples roundtrip" $ examplesRoundtrip ShelleyCodecConfig Shelley.Examples.examplesAllegra
+    , testGroup "Alonzo examples roundtrip"  $ examplesRoundtrip ShelleyCodecConfig Shelley.Examples.examplesAlonzo
+    , testGroup "Babbage examples roundtrip" $ examplesRoundtrip ShelleyCodecConfig Shelley.Examples.examplesBabbage
+    , testGroup "Conway examples roundtrip"  $ examplesRoundtrip ShelleyCodecConfig Shelley.Examples.examplesConway
```

Comment:
Random idea I had: maybe we should store Shelley-based era examples in an `NP Examples ShelleyEras` and map the roundtrip test function over it. That way, we won't forget about adding a roundtrip test when we add a new shelley-based era

Reply:
That's an excellent idea! Thanks. Let me try and see what I can do.